content/en/apps/guides/hosting/4.x/data-migration.md

```diff
@@ -0,0 +1,244 @@
+docker-compose run couch-migration get-env
+```
+
+### 4. Locate and backup CouchDb Data folder
+a) If running in MedicOS, [CouchDb data folder]({{< relref "apps/guides/hosting/3.x/self-hosting#backup" >}}) can be found at `/srv/storage/medic-core/couchdb/data`.
+
+b) If running a custom installation of CouchDb, data would be typically stored at `/opt/couchdb/data`.
+
+c) TODO when using orchestration / AWS 
+
+### 5. Launch 4.x CouchDb installation
+
+Depending on your project scalability needs and technical possibilities, you must decide whether you will deploy CouchDb in a single node or in a cluster with multiple nodes.
+Please consult this guide about clustering and horizontal scalability to make an informed decision. <insert link>
+
+{{% alert title="Note" %}} You can start with single node and then change to a cluster. This involves running the migration tool again to distribute shards from the existent node to the new nodes. {{% /alert %}}
+
+Depending on your choice, follow the instructions that match your deployment below: 
+
+#### Single node
+
+a) Download 4.x single-node CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-single/ 
+cd ~/couchdb-single/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb.yml
+```
+a) Make a copy of the 3.x CouchDb data folder from **step 4**.
+
+b) Set the correct environment variables:
+```shell
+cat > ${HOME}/couchdb-single/.env << EOF
+COUCHDB_USER=<admin>
+COUCHDB_PASSWORD=<password>
+COUCHDB_SECRET=<COUCHDB_SECRET from step 3>
+COUCHDB_UUID=<COUCHDB_UUID from step 3>
+COUCHDB_DATA=<absolute path to folder created in step 5.a>
+EOF
+```
+c) Start 4.x CouchDb and wait until it is up.
+```shell
+cd ~/couchdb-single/ 
+docker-compose up -d
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration check-couchdb-up
+```
+
+d) Change metadata to match the new CouchDb node
+```shell
+cd ~/couchdb-migration/ 
+docker-compose run couch-migration move-node
+```
+e) Verify that the migration was successful
+```shell
+docker-compose run couch-migration verify
+```
+If all checks pass, proceed with starting CHT-Core 4.x, using the same environment variables.   
+
+#### Multi node
+
+a) Download 4.x clustered CouchDb docker-compose file:
+```shell
+mkdir -p ~/couchdb-cluster/ 
+cd ~/couchdb-cluster/ 
+curl -s -o ./docker-compose.yml https://staging.dev.medicmobile.org/_couch/builds_4/medic:medic:4.1.0/docker-compose/cht-couchdb-clustered.yml
+```
+b) Create a data folder for every one of the CouchDb nodes.
+
+c) Copy the 3.x CouchDb data folder into your main CouchDb node's folder. Your main node will be creating your cluster and adding the other secondary nodes to the cluster. You can tell which is your main node by checking your CouchDb docker compose files and identifying which service receives the `CLUSTER_PEER_IPS` environment variable. Your secondary nodes will receive `COUCHDB_SYNC_ADMINS_NODE` variable instead.  
```

Comment:
This seems ambiguous to me:

> You can tell which is your main node by checking your CouchDb docker compose files

Given the importance of this step, I would switch to something like:

```suggestion
c) Copy the 3.x CouchDb data folder into `~/couchdb-cluster/node1`, which will be your main CouchDb node. This main node will create your cluster and the other secondary nodes will be added to it. In `node1`'s environment variable file, define  `CLUSTER_PEER_IPS` . In all other secondary nodes, declare the `COUCHDB_SYNC_ADMINS_NODE` variable instead.  
```

But I'm not sure if that's technically accurate!

Speaking of which - we should specify where you declare these variables.  It seems like each node has it's own `.env` file, but we've only made one in `~/couchdb-cluster/.env`  below. I don't know where to put `CLUSTER_PEER_IPS` vs where to put `COUCHDB_SYNC_ADMINS_NODE` .

Reply:
> It seems like each node has it's own .env file, 

This will depend entirely on how we instruct people to self host clusters, which is now unknown?